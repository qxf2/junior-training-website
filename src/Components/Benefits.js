import React, { Component } from "react";
import Fade from "react-reveal";

class Benefits extends Component {
  render() {
    return (
      <section id="benefits">
        <Fade duration={1000}>
          <div className="row">
            <div className="row">
              <h6
                style={{
                  textAlign: "justify",
                  color: "wheatish",
                  paddingLeft: 50,
                  paddingRight: 50,
                }}
              >
                {" "}
                Why join?
                <ul
                  style={{
                    paddingLeft: 100,
                    fontSize: 14,
                  }}
                >
                  <li>
                    <span
                      style={{
                        color: "green",
                      }}
                    >
                      ✔{" "}
                    </span>
                    How to test software
                  </li>

                  <li>
                    {" "}
                    <span
                      style={{
                        color: "green",
                      }}
                    >
                      ✔{" "}
                    </span>
                    Hands on testing experience
                  </li>
                  <li>
                    {" "}
                    <span
                      style={{
                        color: "green",
                      }}
                    >
                      ✔{" "}
                    </span>
                    Test real applications
                  </li>

                  <li>
                    {" "}
                    <span
                      style={{
                        color: "green",
                      }}
                    >
                      ✔{" "}
                    </span>
                    Participate in Agile rituals
                  </li>

                  <li>
                    {" "}
                    <span
                      style={{
                        color: "green",
                      }}
                    >
                      ✔{" "}
                    </span>
                    Interact with real testers
                  </li>

                  <li>
                    {" "}
                    <span
                      style={{
                        color: "green",
                      }}
                    >
                      ✔{" "}
                    </span>
                    Tools for testing
                    <ul>
                      <li>- GitHub Actions</li>
                      <li>- JavaScript and Python</li>
                      <li>
                        - Test automation -{">"} Selenium, Appium, BrowserStack,
                        API, postman
                      </li>
                      <li>- Test reporting tools - reportportal</li>
                      <li>- Test case management tools</li>
                      <li>- git, GitHub</li>
                      <li>- Jira</li>
                    </ul>
                  </li>

                  <li>
                    <span
                      style={{
                        color: "green",
                      }}
                    >
                      ✔{" "}
                    </span>
                    Interview training
                  </li>

                  <li>
                    {" "}
                    <span
                      style={{
                        color: "green",
                      }}
                    >
                      ✔{" "}
                    </span>
                    Resume reviews
                  </li>

                  <li>
                    {" "}
                    <span
                      style={{
                        color: "green",
                      }}
                    >
                      ✔{" "}
                    </span>
                    Weekly talks (better phrasing needed here)
                  </li>

                  <li>
                    {" "}
                    <span
                      style={{
                        color: "green",
                      }}
                    >
                      ✔{" "}
                    </span>
                    Open source contribution
                  </li>
                </ul>
              </h6>
            </div>
          </div>
        </Fade>
      </section>
    );
  }
}

export default Benefits;
