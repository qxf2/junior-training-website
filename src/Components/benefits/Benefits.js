import React, { Component } from "react";
import Fade from "react-reveal";

class Benefits extends Component {
  render() {
    return (
      <section id="benefits" data-testid="benefits" >
        <Fade duration={1000}>
          <div className="row">
            <Fade bottom>
              <h4>BENEFITS</h4>
            </Fade>
            <div className="twelve columns collapsed">
              <div className="bgrid-quarters s-bgrid-thirds cf">
                <ul>
                  <li>
                    <span
                      style={{
                        color: "green",
                      }}
                    >
                      ✔{" "}
                    </span>
                    Learn to test software
                  </li>

                  <li>
                    {" "}
                    <span
                      style={{
                        color: "green",
                      }}
                    >
                      ✔{" "}
                    </span>
                    Gain hands on testing experience
                  </li>
                  <li>
                    {" "}
                    <span
                      style={{
                        color: "green",
                      }}
                    >
                      ✔{" "}
                    </span>
                    Test real applications
                  </li>

                  <li>
                    {" "}
                    <span
                      style={{
                        color: "green",
                      }}
                    >
                      ✔{" "}
                    </span>
                    Be part of an Agile team
                  </li>

                  <li>
                    {" "}
                    <span
                      style={{
                        color: "green",
                      }}
                    >
                      ✔{" "}
                    </span>
                    Interact, shadow and learn from professional testers
                  </li>

                  <li>
                    {" "}
                    <span
                      style={{
                        color: "green",
                      }}
                    >
                      ✔{" "}
                    </span>
                    Apply modern testing tools and technologies on a daily basis
                    <ul>
                      <li>- JavaScript and/or Python</li>
                      <li>
                        - Test automation -{">"} Selenium and/or Cypress and/or TestCafe, BrowserStack, API tests, Postman
                      </li>
                      <li>- Test reporting using ReportPortal</li>
                      <li>- Test case management with TestRail</li>
                      <li>- Version control with git</li>
                      <li>- Share your code on GitHub</li>
                      <li>- CI/CD with GitHub Actions</li>
                      <li>- Issue tracking with Jira</li>
                    </ul>
                  </li>

                  <li>
                    <span
                      style={{
                        color: "green",
                      }}
                    >
                      ✔{" "}
                    </span>
                    Interview training and mock interviews
                  </li>
                  <li>
                    {" "}
                    <span
                      style={{
                        color: "green",
                      }}
                    >
                      ✔{" "}
                    </span>
                    Detailed resume reviews
                  </li>

                  <li>
                    {" "}
                    <span
                      style={{
                        color: "green",
                      }}
                    >
                      ✔{" "}
                    </span>
                    Weekly talks from industry veterans
                  </li>

                  <li>
                    {" "}
                    <span
                      style={{
                        color: "green",
                      }}
                    >
                      ✔{" "}
                    </span>
                    Guide to becoming an open source contributor
                  </li>
                </ul>
              </div>
            </div>
          </div>
        </Fade>
      </section>
    );
  }
}

export default Benefits;
