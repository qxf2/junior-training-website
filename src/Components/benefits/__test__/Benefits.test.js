import React from 'react';
import ReactDOM from 'react-dom';
import Benefits from '../Benefits';

import { render, cleanup } from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect";

import renderer from "react-test-renderer";

afterEach(cleanup);

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Benefits/>, div);
});

it("renders Benefits correctly", () => {
   const { getByTestId }  = render(<Benefits/>)
   expect(getByTestId('benefits'));
});

it("Benefits.js matches snapshot", () =>{
  const tree = renderer.create(<Benefits/>).toJSON();
  expect(tree).toMatchSnapshot();
});