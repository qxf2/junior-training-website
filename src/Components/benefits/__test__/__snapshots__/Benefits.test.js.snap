// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Benefits.js matches snapshot 1`] = `
<section
  data-testid="benefits"
  id="benefits"
>
  <div
    className="react-reveal row"
    style={
      Object {
        "opacity": undefined,
      }
    }
  >
    <h4
      className="react-reveal"
      style={
        Object {
          "opacity": undefined,
        }
      }
    >
      BENEFITS
    </h4>
    <div
      className="twelve columns collapsed"
    >
      <div
        className="bgrid-quarters s-bgrid-thirds cf"
      >
        <ul>
          <li>
            <span
              style={
                Object {
                  "color": "green",
                }
              }
            >
              ✔
               
            </span>
            Learn to test software
          </li>
          <li>
             
            <span
              style={
                Object {
                  "color": "green",
                }
              }
            >
              ✔
               
            </span>
            Gain hands on testing experience
          </li>
          <li>
             
            <span
              style={
                Object {
                  "color": "green",
                }
              }
            >
              ✔
               
            </span>
            Test real applications
          </li>
          <li>
             
            <span
              style={
                Object {
                  "color": "green",
                }
              }
            >
              ✔
               
            </span>
            Be part of an Agile team
          </li>
          <li>
             
            <span
              style={
                Object {
                  "color": "green",
                }
              }
            >
              ✔
               
            </span>
            Interact, shadow and learn from professional testers
          </li>
          <li>
             
            <span
              style={
                Object {
                  "color": "green",
                }
              }
            >
              ✔
               
            </span>
            Apply modern testing tools and technologies on a daily basis
            <ul>
              <li>
                - JavaScript and/or Python
              </li>
              <li>
                - Test automation -
                &gt;
                 Selenium and/or Cypress and/or TestCafe, BrowserStack, API tests, Postman
              </li>
              <li>
                - Test reporting using ReportPortal
              </li>
              <li>
                - Test case management with TestRail
              </li>
              <li>
                - Version control with git
              </li>
              <li>
                - Share your code on GitHub
              </li>
              <li>
                - CI/CD with GitHub Actions
              </li>
              <li>
                - Issue tracking with Jira
              </li>
            </ul>
          </li>
          <li>
            <span
              style={
                Object {
                  "color": "green",
                }
              }
            >
              ✔
               
            </span>
            Interview training and mock interviews
          </li>
          <li>
             
            <span
              style={
                Object {
                  "color": "green",
                }
              }
            >
              ✔
               
            </span>
            Detailed resume reviews
          </li>
          <li>
             
            <span
              style={
                Object {
                  "color": "green",
                }
              }
            >
              ✔
               
            </span>
            Weekly talks from industry veterans
          </li>
          <li>
             
            <span
              style={
                Object {
                  "color": "green",
                }
              }
            >
              ✔
               
            </span>
            Guide to becoming an open source contributor
          </li>
        </ul>
      </div>
    </div>
  </div>
</section>
`;
